nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/TweetStreamService.java

```diff
@@ -0,0 +1,222 @@
+        this.mediaFields = parseCommaSeparatedPropreties(context, ConsumeTwitter.MEDIA_FIELDS);
+        this.pollFields = parseCommaSeparatedPropreties(context, ConsumeTwitter.POLL_FIELDS);
+        this.placeFields = parseCommaSeparatedPropreties(context, ConsumeTwitter.PLACE_FIELDS);
+        this.expansions = parseCommaSeparatedPropreties(context, ConsumeTwitter.EXPANSIONS);
+        this.backfillMinutes = context.getProperty(ConsumeTwitter.BACKFILL_MINUTES).asInteger();
+
+        this.backoffMultiplier = 1L;
+        this.backoffAttempts = context.getProperty(ConsumeTwitter.BACKOFF_ATTEMPTS).asInteger();
+        this.attemptCounter = 0;
+        this.backoffTime = context.getProperty(ConsumeTwitter.BACKOFF_TIME).asTimePeriod(TimeUnit.SECONDS);
+        this.maximumBackoff = context.getProperty(ConsumeTwitter.MAXIMUM_BACKOFF_TIME).asTimePeriod(TimeUnit.SECONDS);
+
+        ApiClient client = new ApiClient();
+        final int connectTimeout = context.getProperty(ConsumeTwitter.CONNECT_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+        final int readTimeout = context.getProperty(ConsumeTwitter.READ_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+        client = client.setConnectTimeout(connectTimeout);
+        client = client.setReadTimeout(readTimeout);
+        api = new TwitterApi(client);
+
+        TwitterCredentialsBearer creds = new TwitterCredentialsBearer(context.getProperty(ConsumeTwitter.BEARER_TOKEN).getValue());
+        api.setTwitterCredentials(creds);
+
+        this.executorService = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    public String getBasePath() {
+        return api.getApiClient().getBasePath();
+    }
+
+    public void setBasePath(final String path) {
+        api.getApiClient().setBasePath(path);
+    }
+
+    public String getTransitUri(final String endpoint) {
+        if (endpoint.equals(StreamEndpoint.SAMPLE_ENDPOINT.getEndpointName())) {
+            return api.getApiClient().getBasePath() + StreamEndpoint.SAMPLE_ENDPOINT.getPath();
+        } else if (endpoint.equals(StreamEndpoint.SEARCH_ENDPOINT.getEndpointName())) {
+            return api.getApiClient().getBasePath() + StreamEndpoint.SEARCH_ENDPOINT.getPath();
+        } else {
+            logger.warn("Unrecognized endpoint in getTransitUri. Returning basePath");
+            return api.getApiClient().getBasePath();
+        }
+    }
+
+    /**
+     * This method would be called when we would like the stream to get started. This method will spin off a thread that
+     * will continue to queue tweets on to the given queue passed in the constructor. The thread will continue
+     * to run until {@code stop} is called.
+     * @param endpoint {@code TwitterStreamAPI.SAMPLE_ENDPOINT} or {@code TwitterStreamAPI.SEARCH_ENDPOINT}
+     */
+    public void start(final StreamEndpoint endpoint) {
+        this.endpoint = endpoint;
+        executorService.execute(new TweetStreamStarter());
+    }
```

Comment:
As mentioned elsewhere, recommend refactoring to determine the endpoint in the constructor, and simplify this method:
```suggestion
    public void start() {
        executorService.execute(new TweetStreamStarter());
    }
```

Reply:
Making the changes